src/daal4py.cpp

```diff
@@ -232,19 +232,20 @@ extern PyObject * make_nda(daal::data_management::KeyValueDataCollectionPtr * di
 }
 
 template<typename T>
-static daal::data_management::NumericTable * _make_hnt(PyObject * nda)
+static daal::data_management::NumericTablePtr _make_hnt(PyObject * nda)
 {
-    daal::data_management::NumericTable * ptr = NULL;
+    daal::data_management::NumericTablePtr ptr;
     PyArrayObject * array = (PyArrayObject*)nda;
 
     assert(is_array(nda) && array_is_behaved(array));
 
     if(array_numdims(array) == 2) {
         // we provide the SharedPtr with a deleter which decrements the pyref
-        ptr = new daal::data_management::HomogenNumericTable<T>(daal::services::SharedPtr<T>((T*)array_data(array),
-                                                                                             NumpyDeleter(array)),
-                                                                (size_t)array_size(array,1),
-                                                                (size_t)array_size(array,0));
+        ptr = daal::services::dynamicPointerCast<daal::data_management::NumericTable>(
```

Comment:
What is the reason of using `dynamicPointerCast`? You can initialize as follows `ptr=HomogenNumericTable<T>::create(...)`

Reply:
Thanks a lot, fixed in next commit.